Spanner/src/Connection/Grpc.php

```diff
@@ -455,6 +457,39 @@ public function createSession(array $args)
         ]);
     }
 
+    /**
+     * @return PromiseInterface
+     */
+    public function createSessionAsync(array $args)
+    {
+
+        $database = $this->pluck('database', $args);
+        $opts = $this->addResourcePrefixHeader([], $database);
+        $opts['credentialsWrapper'] = $this->credentialsWrapper;
+        $transport = $this->spannerClient->getTransport();
+
+        $request = new CreateSessionRequest([
+            'database' => $database
+        ]);
+
+        $session = $this->pluck('session', $args, false);
+        if ($session) {
+            $sessionMessage = $this->serializer->decodeMessage(new Session, $session);
```

Comment:
Can we just construct a new instance here instead of needing to use the serializer?

Reply:
I was following what the `createSession` did for consistency between these two methods:

```
        $session = $this->pluck('session', $args, false);
        if ($session) {
            $args['session'] = $this->serializer->decodeMessage(new Session, $session);
        }
```

WDYT?

Reply:
Given that context, it's fair you went that route - however since we're using the constructor on lines 471-473 let's go that same route here for consistency. We can address refactoring the rest of the class to not use the serializer as a follow up.

Reply:
👍